Modules/QuestieValidateGameCache.lua

```diff
@@ -0,0 +1,145 @@
+----> Game cache should have all quests in it, data of each quest may be invalid.
+      If data is invalid, Wait for next QUEST_LOG_UPDATE and check again.
+-----> Game Cache ok. Call possible callback functions.
+]]--
+
+---@class QuestieValidateGameCache
+local QuestieValidateGameCache = QuestieLoader:CreateModule("QuestieValidateGameCache")
+
+local stringSub = string.sub
+local GetNumQuestLogEntries, GetQuestLogTitle, GetQuestObjectives = GetNumQuestLogEntries, GetQuestLogTitle, C_QuestLog.GetQuestObjectives
+
+local tpack = table.pack or function(...) return { n = select("#", ...), ... } end
+local tunpack = table.unpack or unpack
+
+-- 3 * (Max possible number of quests in game quest log)
+-- This is a safe value, even smaller would be enough. Too large won't effect performance
+local MAX_QUEST_LOG_INDEX = 75
+
+local eventFrame
+local numberOfQuestLogUpdatesToSkip
+
+local cacheGood = false
+local callbacks = {} -- example: { [1] = {func, {arg1, arg2, arg3}}, [2] = {func, {arg1, arg2}}, }
+
+function QuestieValidateGameCache.IsCacheGood()
+    return cacheGood
+end
+
+---@param func function @A function to call once cache is good.
+---@param ... any @Possible arguments for function.
+function QuestieValidateGameCache.AddCallback(func, ...)
+    callbacks[#callbacks+1] = {func, tpack(...)}
+end
+
+
+local function DestroyEventFrame()
+    if eventFrame then
+        eventFrame:UnregisterAllEvents()
+        eventFrame:SetScript("OnEvent", nil)
+        eventFrame:SetParent(nil)
+        eventFrame = nil
+    end
+end
+
+-- Called directly and OnEvent.
+local function OnQuestLogUpdate()
+    local numEntries, numQuests = GetNumQuestLogEntries()
+    print("--> QuestieValidateGameCache.OnQuestLogUpdate() numEntries:", numEntries, "numQuests:", numQuests)
+
+    if numberOfQuestLogUpdatesToSkip > 0 then
+        numberOfQuestLogUpdatesToSkip = numberOfQuestLogUpdatesToSkip - 1
+        print("--> QuestieValidateGameCache.OnQuestLogUpdate() Skipping event.")
+        return
+    end
+
+    --[[
+    -- Player can have 0 quests in questlog OR game's cache can have empty questlog while cache is invalid
+    -- This is a workaround to wait and see if player really has 0 quests.
+    -- Without cached information the first QLU does not have any quest log entries.
+    -- After MAX_INIT_QUEST_LOG_TRIES tries we stop trying
+    if numEntries == 0 then
+        initQuestLogTries = initQuestLogTries + 1
+        if initQuestLogTries < MAX_INIT_QUEST_LOG_TRIES then
+            print("--> QuestieValidateGameCache.OnQuestLogUpdate()", GetTime(), "Game's quest log is empty. Tries:", initQuestLogTries.."/"..MAX_INIT_QUEST_LOG_TRIES)
+            _WaitForGameCache_TryAgainAtEvent()
+            return
+        else
+            print("--> QuestieValidateGameCache.OnQuestLogUpdate()", GetTime(), "Decide player has no quests for real.")
+        end
+    end
+    ]]--
+
+    local isGameCacheGood = true
```

Comment:
```suggestion
    local isQuestLogFullyCached = true
```
I think this name is better because it is not as easy to confuse with the global `cacheGood`

Reply:
Renamed differently in 78b16dab24437ff7c33ec1111cd51a01e38c7ea2